### PR TITLE
Add pio pwm example

### PR DESCRIPTION
**Pull request by hmvp**
This adds a more advance pio example which uses side set and instruction injection

### CODE REVIEW

**Comment by hmvp** [comment]
Moved pio code to seperate file so we can also close  #154

File: boards/rp-pico/examples/pico_pio_pwm.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+/// This uses a sneaky trick to set a second value besides the duty cycle.
+/// We first write a value to the tx fifo. But instead of the normal instructions we
+/// have stopped the state machine and inject our own instructions that move the written value to the ISR.
+fn pio_pwm_set_period<T: ValidStateMachine>(
+    sm: StateMachine<(hal::pac::PIO0, SM0), Running>,
+    tx: &mut Tx<T>,
+    period: u32,
+) -> StateMachine<(hal::pac::PIO0, SM0), Running> {
+    let mut sm = sm.stop();
+    tx.write(period);
```
Comment (jannic) [inline boards/rp-pico/examples/pico_pio_pwm.rs:54]: Calling `pio_pwm_set_period` shortly after `pio_pwm_set_level` could have unintended results: If the value written to the tx fifo by `pio_pwm_set_level` has not yet been read by the PIO program before the state machine is stopped, the level will be interpreted as a period.

Perhaps add `while !tx.is_empty() {}` before the call to `sm.stop()`?

Comment (hmvp) [reply boards/rp-pico/examples/pico_pio_pwm.rs:54]: Interesting.. You are completely right. I based the example on https://github.com/raspberrypi/pico-examples/blob/master/pio/pwm/pwm.c which has the same issue...

**Review by 9names — APPROVED** [review]
LGTM

**Pull request merged**